test/api/advanced/find-or-create.test.js

```diff
@@ -151,3 +151,81 @@ test('advanced.findOrCreate (protected metadata fields)', (t) => {
     )
   })
 })
+
+test('advanced.findOrCreate (encryption - GroupId)', (t) => {
+  const sbot = Testbot()
+
+  const groupId = '%EPdhGFkWxLn2k7kzthIddA8yqdX8VwjmhmTes0gMMqE=.cloaked'
+  const groupKey = Buffer.from(
+    '30720d8f9cbf37f6d7062826f6decac93e308060a8aaaa77e6a4747f40ee1a76',
+    'hex'
+  )
+  sbot.box2.addGroupKey(groupId, groupKey)
+
+  testReadAndPersisted(t, sbot, (t, sbot, cb) => {
+    const series = [
+      (cb) => sbot.metafeeds.advanced.findOrCreate(cb),
+      (mf, cb) => {
+        sbot.metafeeds.advanced.findOrCreate(
+          mf,
+          (f) => f.feedpurpose === 'private',
+          {
+            feedpurpose: 'private',
+            feedformat: 'classic',
+            recps: [groupId],
+            encryptionFormat: 'box2'
+          },
+          cb
+        )
+      },
+      (f, cb) => {
+        t.deepEqual(f.recps, [groupId], 'FeedDetails contains recps')
+
+        sbot.db.query(where(type('metafeed/add/derived')), toCallback((err, msgs) => {
+          if (err) return cb(err)
+
+          if (msgs.length !== 1) t.equal(msgs.length, 1, 'only one metafeed/add/derived')
+
+          t.deepEqual(msgs[0].value.content.recps, [groupId], 'metafeed/add/derived has recps')
+          // t.deepEqual(typeof msgs[0].value?.meta?.originalContent, 'string', 'metafeed/add/derived is encrypted')
+
+          cb(null)
+        }))
+      }
+    ]
+
+    waterfall(series, cb)
```

Comment:
Could we not add this dependency? I want the tests to run only "raw JS" code and *our* library, to rule out mistakes from other libraries and test only our library. If I understand correctly, we can use just a traditional callback pyramid for this purpose. It's not pretty, but everyone understands what's going on, unlike a new dependency.

Reply:
The pyramid is annoying to maintain IMO.
Agree we should discuss "standard tools" we're using. Will bring this up in next meeting and change this for now

Reply:
DONE